app/script/localization/Localizer.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Wire
+ * Copyright (C) 2017 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+z.localization = z.localization || {};
+
+class Localizer {
+  constructor() {
+    let param = z.util.get_url_parameter(z.auth.URLParameter.LOCALE);
+    if (param) {
+      z.util.StorageUtil.set_value(z.storage.StorageKey.LOCALIZATION.LOCALE, param);
+    }
+    this.locale = z.util.StorageUtil.get_value(z.storage.StorageKey.LOCALIZATION.LOCALE) || navigator.language.substr(0, 2) || 'en';
+    moment.locale([this.locale, 'en']);
+    if (z.string[this.locale]) {
+      $.extend(z.string, z.string[this.locale]);
+    }
+  }
+
+  /*
+  Pulls the localized string from the resources and replaces placeholders.
+
+  @note Takes the id of the string for look up from z.string is directly for simple use. Else pass it in as the id
+    parameter in conjunction with a single or multiple (it supports but does not require an array) replace rules that
+    consist of a placeholder and the content that it should be replace with.
+
+  @param id [String] Localization string ID
+  @param replace [Object | Array<Object>] Placeholders that should be replaced
+  @option replace [String] placeholder Content to be replaced
+  @option replace [String] content replacing content
+    */
+
+  get_text(valueAccessor) {
+    if (valueAccessor == null) return;
+
+    let args = [];
+    let s = valueAccessor;
```

Comment:
There is no `@param id` nor a `@param replace`. I only see `@param valueAccessor`.